### PR TITLE
Dhruv2000/schema validation

### PR DESCRIPTION
**Pull request by dhruv2000**
Needs review of Joi Schema Validation. Might have some syntactical errors.

### CODE REVIEW

File: src/models/schemaValidation.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+const Joi = require('@hapi/joi');
```
Comment (marcgabe15) [inline src/models/schemaValidation.ts:1]: I think this package is deprecated https://www.npmjs.com/package/@hapi/joi Use joi instead of @hapi/joi

Comment (marcgabe15) [inline src/models/schemaValidation.ts:1]: Also do we need the types file for this? (d.ts file)

**Review by marcgabe15 — CHANGES_REQUESTED** [review]

**Review by HamishPierpont — COMMENTED** [review]
This looks good, but the model code ended up in Marc's more recent pull requests and the formatting is slightly different in his because he added `prettier` style linting to the repository. If we merge this one and we also merge Marc's PR(s), I think we'll end up with unnecessary merge conflicts. We also decided in our last meeting that we wanted to follow the layout of the boilerplate [repo](https://github.com/danielfsousa/express-rest-boilerplate/tree/9e2189f0067c416a09d951c95188bdf2778005ad/src/api/validations). We still want to get Dhruv's schema validation code into the repo, but we'll want to do it with the new linter and with the file and folder style of the boilerplate repo. Here's what I think we should do:
1. Merge the most recent PR from Marc into `main`, which I believe contains the model code, linting, and the health check route.
2. Dhruv should save his `schemaValidation.ts` file and then close this PR.
3. Dhruv should make a new branch off the updated `main` repository, add his validation code to the repository as `src/validations/models.validation.ts`, let the linter format it correctly, and then submit a new PR to merge the code back into the repo with that formatting.

Let me know what you guys think of this plan.

**Comment by marcgabe15** [comment]
> This looks good, but the model code ended up in Marc's more recent pull requests and the formatting is slightly different in his because he added `prettier` style linting to the repository. If we merge this one and we also merge Marc's PR(s), I think we'll end up with unnecessary merge conflicts. We also decided in our last meeting that we wanted to follow the layout of the boilerplate [repo](https://github.com/danielfsousa/express-rest-boilerplate/tree/9e2189f0067c416a09d951c95188bdf2778005ad/src/api/validations). We still want to get Dhruv's schema validation code into the repo, but we'll want to do it with the new linter and with the file and folder style of the boilerplate repo. Here's what I think we should do:
> 
> 1. Merge the most recent PR from Marc into `main`, which I believe contains the model code, linting, and the health check route.
> 2. Dhruv should save his `schemaValidation.ts` file and then close this PR.
> 3. Dhruv should make a new branch off the updated `main` repository, add his validation code to the repository as `src/validations/models.validation.ts`, let the linter format it correctly, and then submit a new PR to merge the code back into the repo with that formatting.
> 
> Let me know what you guys think of this plan.

This makes sense and yes I agree

**Pull request closed without merging**